### PR TITLE
[CALCITE-3607] Support LogicalTableModify in RelShuttle and fix typos

### PR DESCRIPTION
**Pull request by xy2953396112**
RelShuttle don't support LogicalTableModify,this pr try to support LogicalTableModify in RelShuttle.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 1]
```diff
@@ -1987,6 +1989,21 @@ protected final void check(
...
   }
 
+  @Test public void testRelShuttleForLogicalTableModify() {
+    String sql = "insert into emp select * from emp";
+    LogicalTableModify rel = (LogicalTableModify) tester.convertSqlToRel(sql).rel;
+    List<RelNode> rels = new ArrayList<>();
+    final RelShuttleImpl visitor = new RelShuttleImpl() {
+      @Override public RelNode visit(LogicalTableModify modify) {
+        rels.add(modify);
+        return super.visit(modify);
```
Comment (amaliujia) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java]: I am thinking the right test to the code should be:

```
@Override public RelNode visit(LogicalTableModify modify) {
      Rel visitedRel = super.visit(modify);        
      rels.add(visitedRel);
       return visitedRel;
}

```

Comment (danny0405) [reply core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java]: For this test they are equivalent, but i think you are right

**Comment by amaliujia** [comment]
LGTM

**Comment by xy2953396112** [comment]
@amaliujia @danny0405 Thanks for you review.

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 2]
```diff
@@ -1987,6 +1989,22 @@ protected final void check(
     assertTrue(filterCollation.satisfies(sortCollation));
   }
 
+  @Test public void testRelShuttleForLogicalTableModify() {
+    String sql = "insert into emp select * from emp";
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java]: final

**Comment by jinxing64** [comment]
+1

**Comment by xy2953396112** [comment]
@danny0405 @chunweilei  Thanks for you review, merge [fix typos] (https://github.com/apache/calcite/pull/1700)

**Pull request closed without merging**